code/socialDistribution/views.py

```diff
@@ -436,65 +436,62 @@ def share_post(request, id):
+        
+        form = PostForm(request.POST, request.FILES, user_id=post.author.id)
         if form.is_valid():
             bin_content = form.cleaned_data.get('content_media')
+            content_media = None
             if bin_content is not None:
-                content_media = base64.b64encode(bin_content.read())
+                content_media = base64.b64encode(bin_content.read()).decode('ascii')
             else:
                 content_media = post.content_media
 
             try:
-                post.title = form.cleaned_data.get('title')
-                post.description = form.cleaned_data.get('description')
-                post.content = form.cleaned_data.get('content_text')
-                post.visibility = form.cleaned_data.get('visibility')
-                post.unlisted = form.cleaned_data.get('unlisted')
-                post.content_media = content_media
+                edited_post['title'] = form.cleaned_data.get('title')
+                edited_post['description'] = form.cleaned_data.get('description')
+                edited_post['content'] = form.cleaned_data.get('content_text')
+                edited_post['visibility'] = form.cleaned_data.get('visibility')
+                edited_post['unlisted'] = form.cleaned_data.get('unlisted')
+                edited_post['content_media'] = content_media
 
                 categories = form.cleaned_data.get('categories')
 
-                if categories is not None:
-                    categories = categories.split()
-                    categories_to_remove = [ cat.category for cat in post.categories.all()]
-
-                    """
-                    This implementation makes category names case-insensitive.
-                    This makes handling Category objects cleaner, albeit slightly more
-                    involved.
-                    """
-                    for category in categories:
-                        category_obj, created = Category.objects.get_or_create(
-                            category__iexact=category,
-                            defaults={'category': category}
-                        )
-                        post.categories.add(category_obj)
-                        
-                        while category_obj.category in categories_to_remove:
-                            categories_to_remove.remove(category_obj.category)     # don't remove this category
-
-                    for category in categories_to_remove:
-                        category_obj = Category.objects.get(category=category)
-                        post.categories.remove(category_obj)
-
-                post.save()
-
+                edited_post['categories'] = categories.split()
+                
             except ValidationError:
                 messages.info(request, 'Unable to edit post.')
+                
+        # redirect request to remote/local api
+        response = make_request('POST', request_url, json.dumps(edited_post))
```

Comment:
I think this request_url will require user authentication. Since we only ever edit local posts it might be best to keep the logic in this function and not use an API call for it.

I think you should keep code for the API routes that you worked on, because we do still need them for the API spec, but don't call the API from here because you won't be able to authenticate. 

Let me know if that makes sense or if you disagree.

Reply:
That sounds good